test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -88,6 +88,30 @@ public void Controller_View_WithoutParameter_SetsResultNullViewNameAndNullViewDa
         }
 
         [Fact]
+        public void Controller_View_WithoutParameter_MaintainsModelInViewData()
+        {
+            // Arrange
+            var controller = new TestableController()
+            {
+                ViewData = new ViewDataDictionary(new EmptyModelMetadataProvider()),
+                TempData = new TempDataDictionary(new DefaultHttpContext(), Mock.Of<ITempDataProvider>()),
+            };
+
+            var model = new object();
+            controller.ViewData.Model = model;
+
+            // Act
+            var actualViewResult = controller.View();
```

Comment:
Should also test:
- `controller.PartialView()`

Less sure whether it's worth confirming the other overloads:
- `controller.View(viewName: "something")`
- `controller.PartialView(viewName: "something")`